internal/cloud/awscloud/autoscaling.go

```diff
@@ -0,0 +1,45 @@
+package awscloud
+
+import (
+	"context"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/feature/ec2/imds"
+	"github.com/aws/aws-sdk-go-v2/service/autoscaling"
+)
+
+func (a *AWS) ASGSetProtectHost(protect bool) error {
+	identity, err := a.ec2imds.GetInstanceIdentityDocument(context.Background(), &imds.GetInstanceIdentityDocumentInput{})
+	if err != nil {
+		return err
+	}
+
+	descrASG, err := a.asg.DescribeAutoScalingInstances(
+		context.Background(),
+		&autoscaling.DescribeAutoScalingInstancesInput{
+			InstanceIds: []string{
+				identity.InstanceID,
+			},
+		},
+	)
+	if err != nil {
+		return nil
```

Comment:
Do we want to return nil here? Or should this be `return err` ? If `return nil` maybe a comment why?

Reply:
Looked at it again, and at this point we'd know we're running in AWS, but not that we're running as part of an ASG. So let's return the error and let the worker log an info message

```
...
		logrus.Infof("Unable to protect host, if the host isn't running as part of an autoscaling group, this can safely be ignored: %v", err)
```